### PR TITLE
release-23.2: changefeedccl: add new table bytes metric for usage data

### PR DESCRIPTION
**Pull request by asg0451**
Backport 1/1 commits from #123145.

/cc @cockroachdb/release

---

Add a new metric to facilitate the new changefeed billing strategy - `changefeed.usage.table_bytes` - This is a sum of the bytes watched per changefeed. It intentionally counts tables watched by multiple changefeeds multiple times. Also add auxiliary metrics about usage data collection: `changefeed.usage.error_count` (counter of errors encountered running the query that generates the table_bytes metric) and `changefeed.usage.query_duration` (histogram of how long the latter took). The collection of this data and emission of these metrics are gated behind an environment variable only intended to be set in ccloud: `COCKROACH_ENABLE_CLOUD_BILLING_ACCOUNTING`

Epic: CRDB-37031
Resolves: #121781

Release note (enterprise change): Add a new metric to facilitate the new changefeed billing strategy for usage in Cockroach Cloud.

Release justification: This is a necessary change for upcoming billing changes.

### CODE REVIEW

File: pkg/sql/plpgsql/parser/BUILD.bazel [hunk 1]
```diff
@@ -35,6 +35,7 @@ go_library(
     importpath = "github.com/cockroachdb/cockroach/pkg/sql/plpgsql/parser",
     visibility = ["//visibility:public"],
     deps = [
+        "//pkg/build",
```
Comment (asg0451) [inline pkg/sql/plpgsql/parser/BUILD.bazel:38]: idk why this change was made by `dev gen`...

Comment (rharding6373) [reply pkg/sql/plpgsql/parser/BUILD.bazel:38]: From the failed run it looks like it needs to run dev gen bazel again. Maybe that will help?

**Review by rgcase — APPROVED** [review]

**Review by rharding6373 — APPROVED** [review]
